pkg/names/generate.go

```diff
@@ -52,3 +56,10 @@ func (simpleNameGenerator) GenerateName(base string) string {
 	}
 	return fmt.Sprintf("%s-%s", base, utilrand.String(randomLength))
 }
+
+func (simpleNameGenerator) GenerateBuildStepName(base string) string {
```

Comment:
i think this name reflects how this function is being used, but maybe we could name it after what it's doing? It seems like the functionality we'll have is something like:

- `GenerateName` - appends a random prefix AND truncates a string if it's too long (maybe call this `RestrictLengthWithRandomPrefix` - called as `names.RestrictLengthWithRandomPrefix`
- `GenerateBuildStepName` - truncates a string if it's too long (name idea: `RestrictLength` -  since it'd be called as `names.RestrictLength`)

Reply:
Done.